packages/apps/reputation-oracle/server/src/modules/abuse/abuse.service.spec.ts

```diff
@@ -0,0 +1,495 @@
+import { faker } from '@faker-js/faker';
+import { createMock } from '@golevelup/ts-jest';
+import { EscrowUtils, OperatorUtils, StakingClient } from '@human-protocol/sdk';
+import { ConfigService } from '@nestjs/config';
+import { Test } from '@nestjs/testing';
+import { EventType, ReputationEntityType } from '../../common/enums';
+import { PostgresErrorCodes } from '../../common/enums/database';
+import { DatabaseError } from '../../common/errors/database';
+import { ServerConfigService } from '../../config/server-config.service';
+import { ReputationService } from '../reputation/reputation.service';
+import { Web3Service } from '../web3/web3.service';
+import { WebhookOutgoingService } from '../webhook/webhook-outgoing.service';
+import { AbuseRepository } from './abuse.repository';
+import { AbuseService } from './abuse.service';
+import { AbuseSlackBot } from './abuse.slack-bot';
+import { AbuseDecision, AbuseStatus } from './constants';
+import { generateAbuseEntity } from './fixtures';
+
+const fakeAddress = faker.finance.ethereumAddress();
+jest.mock('@human-protocol/sdk', () => ({
+  ...jest.requireActual('@human-protocol/sdk'),
+  StakingClient: {
+    build: jest.fn().mockImplementation(() => ({
+      slash: jest.fn(),
+    })),
+  },
+  OperatorUtils: {
+    getOperator: jest.fn(),
+  },
+  EscrowUtils: {
+    getEscrow: jest.fn(),
+  },
+}));
+
+describe('AbuseService', () => {
+  let abuseService: AbuseService;
+  let abuseSlackBot: AbuseSlackBot;
+  let abuseRepository: AbuseRepository;
+  let webhookOutgoingService: WebhookOutgoingService;
+  let reputationService: ReputationService;
+
+  const escrowAddress = faker.finance.ethereumAddress();
+  const chainId = faker.number.int({ max: 100000 });
+  const webhookUrl1 = faker.internet.url();
+  const webhookUrl2 = faker.internet.url();
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [
+        AbuseService,
+        ConfigService,
+        ServerConfigService,
+        {
+          provide: AbuseSlackBot,
+          useValue: createMock<AbuseSlackBot>(),
+        },
+        {
+          provide: Web3Service,
+          useValue: {
+            getSigner: jest.fn().mockReturnValue({
+              address: faker.finance.ethereumAddress(),
+              getNetwork: jest.fn().mockResolvedValue({ chainId }),
+            }),
+          },
+        },
+        { provide: AbuseRepository, useValue: createMock<AbuseRepository>() },
+        {
+          provide: ReputationService,
+          useValue: createMock<ReputationService>(),
```

Comment:
Please defined all mocks after imports, you don't have to get them from compiled module

It's a "style"-thing, but makes it easier to digest unit tests files when you looking at them